code/modules/mob/living/basic/lavaland/goldgrub/goldgrub_abilities.dm

```diff
@@ -0,0 +1,53 @@
+/datum/action/cooldown/mob_cooldown/spit_ore
+	name = "Spit Ore"
+	desc = "Vomit out all of your consumed ores."
+	click_to_activate = FALSE
+	cooldown_time = 5 SECONDS
+
+/datum/action/cooldown/mob_cooldown/spit_ore/Activate()
+	var/mob/living/basic/mining/goldgrub/grub_owner = owner
+	grub_owner.barf_contents()
+	StartCooldown()
+	return TRUE
```

Comment:
Should fail if there's no ore
```dm
if(!length(grub_owner.contents)
    grub_owner.balloon_alert(grub_owner, "no ores to spit!"_
    return FALSE
```
(If you want to go the extra mile you can put this in `IsAvailable` and register a `update_status_on_signal` for contents entering or exiting the owner)